spec/lang/types.md

```diff
@@ -189,5 +189,17 @@ impl IntType {
     pub fn bring_in_bounds(&self, i: Int) -> Int {
         i.bring_in_bounds(self.signed, self.size)
     }
+
+    /// Generate the return type for IntWithOverflow
+    pub fn overflow_type<T: Target>(&self) -> Type {
+        // Define a tuple type with two fields: An integer followed directly by a boolean.
+        let self_ty = Type::Int(*self);
+        let fields = list![(Size::ZERO, self_ty), (self_ty.size::<T>(), Type::Bool)];
+        // Alignment is always the one of the integer as boolean has align requirement 1.
+        let align = self_ty.align::<T>();
+        // As size is multiple of alignment boolean occupies space of integer size.
+        let size = self_ty.size::<T>() * Int::from(2);
```

Comment:
This is actually wrong when the integer has size != align, isn't it? For instance, `u64` on a 32-bit-target works like that.

Reply:
Oh, I did not know this is possible. I changed it such that we now add size and alignment.